.github/workflows/terraform-apply.yml

```diff
@@ -0,0 +1,42 @@
+name: Terraform apply
+
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - 'config/clusters/**'
+
+defaults:
+  run:
+    working-directory: config/clusters
+
+env:
+  AWS_REGION: 'eu-west-1'
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+jobs:
+  plan:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: hashicorp/setup-terraform@v2
+    
+    - name: Terraform fmt
+      id: fmt
+      run: terraform fmt -check
+      continue-on-error: true
+    
+    - name: Terraform Init
+      id: init
+      run: terraform init
+    
+    - name: Terraform Validate
+      id: validate
+      run: terraform validate -no-color
+    
+    - name: Terraform Apply
+      id: apply
+      run: terraform apply -auto-approve
```

Comment:
This is not how should be done, as the desired state (the TF plan) is re-calculated, during this workflow, which is triggered by new commits on the master branch, when changes affect the infrastructure-code.
Documentation: https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform#interactive-approval-of-plans.

Nevertheless, I think it's a valid trade-off, considering that this specific GitHub event doesn't link to a PR AFAIK, and it would be hard to withdraw the correct plan which was generated in the respective approved PR.

Reply:
The current master gha leverages the `concurrency` keyword: https://github.com/falcosecurity/test-infra/blob/master/.github/workflows/master.yml#L7.
This allows us to enqueue subsequent master CI jobs until each one of them finishes (in a FIFO manner).

Reply:
We are not running them concurrently, we are using the `concurrency` group to create a "queue" of master deployments, so that indeed we never run them concurrntly.

Reply:
> We are not running them concurrently, we are using the `concurrency` group to create a "queue" of master deployments, so that indeed we never run them concurrntly.

Oh ok thanks, I misunderstood about what would be queued.
Then I agree, I'm going to update it.
In addition to this barrier, theres a write lock during apply, so we're safe I guess :)

Reply:
Added @FedeDP, feel free to resolve this conversation